### ISSUE TITLE
aria-required-children false positive for almost empty nodes

### ISSUE THREAD

**Issue by WilcoFiers** [dequelabs/axe-core#1762]
The `aria-required-children` rule has a `reviewEmpty` option, which tells axe-core what roles to fail when they are empty, and which ones not to. For example, `listbox` is allowed to be empty.

```html
<div role="listbox" aria-label="Select nothing"></div>
```

That element shouldn't fail. However, if we put something innocuous in it, like this:

```html
<div role="listbox" aria-label="Select nothing">
  <div></div>
</div>
```

Axe-core fails the element. This happens because axe-core only looks if there are any children at all. I think axe-core needs to be a bit smarter about when to consider an item "empty".

I think the best way to do this, is to see if there are any "owned elements". I'm not entirely sure about the best way to solve this. We could maybe look at the owned elements and check if they have a role or an accessible name? That may create other gaps.

**Comment by straker** [comment]
At what point does the element become non-empty? If it has a role but no text content? When it has just text content? What about implicit role elements like `<nav>`? Or if the `<nav>` element has an `aria-label` on it but no text content?

**Comment by jeankaplansky** [comment]
@straker : technically, an element becomes non-empty when you add any text or markup within the start and end tags (XML definition). Attributes aren't considered content. Here's Mozilla's official: https://developer.mozilla.org/en-US/docs/Glossary/Empty_element

**Comment by chandana7393** [comment]
Tested, working as expected.
![aria_role_children](https://user-images.githubusercontent.com/45285854/64336685-32f26c80-cffb-11e9-88e0-8d4d07fae99a.PNG)

**Tested Environment:**
Attest - 2.6.1.24714v
Axe - 3.9.0.24714v
Axe-coconut - 3.9.0.24804v
Chrome - 76.0.3809.132v
Firefox - 68.0.2v
OS - Windows 10 64 bit.
